interfaces/cython/cantera/onedim.py

```diff
@@ -570,19 +570,10 @@ def __init__(self, gas, grid=None, width=None):
         super(IonFlame, self).__init__(gas, grid, width)
 
     def solve(self, loglevel=1, refine_grid=True, auto=False, stage=1, enable_energy=True):
-        if enable_energy == True:
-            self.energy_enabled = True
-            self.velocity_enabled = True
-        else:
-            self.energy_enabled = False
-            self.velocity_enabled = False
         if stage == 1:
             self.flame.set_solvingStage(stage)
             super(IonFlame, self).solve(loglevel, refine_grid, auto)
         if stage == 2:
-            self.flame.set_solvingStage(stage)
-            super(IonFlame, self).solve(loglevel, refine_grid, auto)
-        if stage == 3:
             self.flame.set_solvingStage(stage)
```

Comment:
The bug here with still calling `stage=3` could be fixed by moving the call `self.flame.set_solvingStage(stage)` out of the `if` statement.